api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/TenableAccountServiceImpl.java

```diff
@@ -77,75 +75,98 @@ private void validateAccountCredentials(CreateAccountRequest accountData, Accoun
-            validateResponse.setMessage("Tenable validation Failed");
         } catch (IOException e) {
             LOGGER.error("Failed to validate the tenable account ", e.getMessage());
-            validateResponse.setValidationStatus(FAILURE);
-            validateResponse.setMessage("Tenable validation Failed: " + e.getMessage());
+            validationResponse.setValidationStatus(FAILURE);
+            validationResponse.setMessage("Tenable account validation failed");
         }
+
+        return validationResponse;
     }
 
     @Override
     public AccountValidationResponse addAccount(CreateAccountRequest accountData) {
         LOGGER.info("Adding new Tenable account....");
-        AccountValidationResponse validateResponse= validateRequestData(accountData);
-        if(validateResponse.getValidationStatus().equalsIgnoreCase(FAILURE)){
-            LOGGER.info("Validation failed due to missing parameters");
+        AccountValidationResponse validateResponse;
+
+        if (getListAccountsByPlatform(TENABLE).size() > 0) {
+            LOGGER.info("Tenable account already exists");
+            validateResponse = new AccountValidationResponse();
+            validateResponse.setValidationStatus(FAILURE);
+            validateResponse.setMessage("Tenable account already exists");
+            return validateResponse;
+        }
+
+        validateResponse = validate(accountData);
+        if (validateResponse.getValidationStatus().equalsIgnoreCase(FAILURE)) {
+            LOGGER.info("Adding account failed: {}", validateResponse.getMessage());
             return validateResponse;
         }
         BasicSessionCredentials credentials = credentialProvider.getBaseAccCredentials();
         String region = System.getenv("REGION");
-        String roleName= System.getenv(PALADINCLOUD_RO);
 
         AWSSecretsManager secretClient = AWSSecretsManagerClientBuilder
                 .standard()
                 .withCredentials(new AWSStaticCredentialsProvider(credentials))
                 .withRegion(region).build();
 
-        CreateSecretRequest createRequest=new CreateSecretRequest()
-                .withName(secretManagerPrefix+ "/" + roleName + "/tenable").withSecretString(getTenableSecret(accountData));
+        String secretId = getTenableSecretId();
+        ListSecretsRequest listSecretsRequest = new ListSecretsRequest()
+                .withFilters(new Filter().withKey("name").withValues(secretId))
+                .withIncludePlannedDeletion(true);
+        ListSecretsResult getListResponse = secretClient.listSecrets(listSecretsRequest);
+
+        if (getListResponse.getSecretList().size() > 0) {
+            LOGGER.info("Secret already exists. Deleting in progress.");
+            validateResponse = new AccountValidationResponse();
+            validateResponse.setValidationStatus(FAILURE);
+            validateResponse.setMessage("Similar account is being deleted. Please try again later.");
```

Comment:
Lets rephrase this to:
We are currently processing the deletion of a prior account. Please try your request again at a later time.

Reply:
I think it's too big for simple notification in UI. Let's have it as "Prior account is being deleted. Try again in a couple of minutes."